src/ImageSharp/Common/Memory/PixelDataPool{T}.cs

```diff
@@ -9,39 +9,69 @@ namespace ImageSharp
     using System.Buffers;
 
     /// <summary>
-    /// Provides a resource pool that enables reusing instances of value type arrays <see cref="T:T[]"/>.
-    /// <see cref="Rent(int)"/> will always return arrays initialized with 'default(T)'
+    /// Provides a resource pool that enables reusing instances of value type arrays for image data <see cref="T:T[]"/>.
     /// </summary>
     /// <typeparam name="T">The value type.</typeparam>
-    public static class PixelDataPool<T>
+    public class PixelDataPool<T>
         where T : struct
     {
         /// <summary>
-        /// The <see cref="ArrayPool{T}"/> used to pool data.
+        /// The <see cref="ArrayPool{T}"/> which will be always cleared.
         /// </summary>
-        private static readonly ArrayPool<T> ArrayPool = ArrayPool<T>.Create(CalculateMaxArrayLength(), 50);
+        private static readonly ArrayPool<T> CleanPool = ArrayPool<T>.Create(CalculateMaxArrayLength(), 50);
+
+        /// <summary>
+        /// The <see cref="ArrayPool{T}"/> which is not kept clean.
+        /// </summary>
+        private static readonly ArrayPool<T> DirtyPool = ArrayPool<T>.Create(CalculateMaxArrayLength(), 50);
+
+        /// <summary>
+        /// The backing <see cref="ArrayPool{T}"/>
+        /// </summary>
+        private ArrayPool<T> arrayPool;
+
+        /// <summary>
+        /// A value indicating whether clearArray is requested on <see cref="ArrayPool{T}.Return(T[], bool)"/>.
+        /// </summary>
+        private bool clearArray;
+
+        private PixelDataPool(ArrayPool<T> arrayPool, bool clearArray)
+        {
+            this.clearArray = clearArray;
+            this.arrayPool = arrayPool;
+        }
+
+        /// <summary>
+        /// Gets the <see cref="PixelDataPool{T}"/> which will always return arrays initialized to default(T)
+        /// </summary>
+        public static PixelDataPool<T> Clean { get; } = new PixelDataPool<T>(CleanPool, true);
```

Comment:
why do we need arrays defaulted to default(T) at all? Do we have logic that is dependent on the data being default? if we do how many places? could we not just reset them ourselves if required and save the added complexity of having more than one data pool? 

My guess is this is for the empty buffer for `new Image(height, width)` could we not just add a simple T[].Clear() extension method, or `RentClean(int minimumLength)` method that can be called to set the default value if required. I feel creating a new images is probably going to happen less than opening an existing file which shouldn't need a clear array anyway as it will be setting all the pixels itself.